internal/bulk.go

```diff
@@ -42,8 +42,12 @@ type BulkResponse struct {
 	IsFinal bool `json:"is_final"`
 }
 
+type BulkOptions struct {
+	DefaultPrivateKey *dsig.PrivateKey
+}
+
 // Bulk processes a stream of bulk requests.
-func Bulk(ctx context.Context, in io.Reader) <-chan *BulkResponse {
+func Bulk(ctx context.Context, opts *BulkOptions, in io.Reader) <-chan *BulkResponse {
```

Comment:
Semi-convention in Go would put the `opts` at the end of the parameters, or even expect the `in` parameter to be included in the options. Most people would have probably just added the private key to the `Bulk` operation directly :-)

Reply:
Thanks for the input! I like the idea of making `in` part of the options. I'll give it a try.

Reply:
Amended in my last push